### PR TITLE
BE-221: add region manage view

### PR DESCRIPTION
**Pull request by liumengjun**

### CODE REVIEW

**Comment by shangerdi** [comment]
LGTM

File: server/app/models.py [hunk 1]
```diff
@@ -1568,7 +1568,7 @@ class WeeklyTimeSlot(BaseModel):
     end = models.TimeField()
 
     class Meta:
-        unique_together = ('weekday', 'start', 'end')
+        # unique_together = ('weekday', 'start', 'end')
```
Comment (1e0ng) [inline server/app/models.py]: Just remove this line as this will never be used.

File: server/staff/urls.py [hunk 2]
```diff
@@ -37,4 +37,6 @@
     url(r'^school/account/info/$', views.SchoolAccountInfoView.as_view(), name="school_account_info"),
     url(r'^school/price/config/$', views.SchoolPriceConfigView.as_view(), name="school_price_cfg"),
     url(r'^school/account/income/records/$', views.SchoolIncomeRecordView.as_view(), name="school_income_records"),
+    url(r'^region/(?P<rid>\d+)/$', views.RegionView.as_view(), name="region_show"),
+    url(r'^region/(?P<rid>\d+)/(?P<action>open)/$', views.RegionView.as_view(), name="region_open"),
```
Comment (1e0ng) [inline server/staff/urls.py]: It's better to move the action `open` into the query parameter like `https://.../region/(?P<rid>\d+)/?action=open`, so only one url rule is sufficient. After all, url should describe a resource, not actions.

File: server/staff/views.py [hunk 3]
```diff
@@ -2745,3 +2745,33 @@ def get_context_data(self, **kwargs):
...
+        if action == 'open':
+            return self._open_city(region)
+
+        html_buf = []
+        html_buf.append(region.name)
+        for wts in region.weekly_time_slots.all():
+            html_buf.append(str(wts))
+        return HttpResponse('<br>'.join(html_buf))
+
+    def _open_city(self, region):
```
Comment (1e0ng) [inline server/staff/views.py:2798]: Could a city be opened multiple times? If that happens, there may be junk data, right?

File: server/app/models.py (hunk 1 above)
Comment (liumengjun) [reply server/app/models.py]: done

File: server/staff/urls.py (hunk 2 above)
Comment (liumengjun) [reply server/staff/urls.py]: done

File: server/staff/views.py (hunk 3 above)
Comment (liumengjun) [reply server/staff/views.py:2798]: it was checked

**Comment by 1e0ng** [comment]
This branch has conflicts that must be resolved

**Comment by liumengjun** [comment]
lgtm

**Pull request merged**